crates/steel-core/src/steel_vm/engine.rs

```diff
@@ -1404,13 +1404,10 @@ impl Engine {
         fn eval_atom(t: &SyntaxObject) -> Result<SteelVal> {
             match &t.ty {
                 TokenType::BooleanLiteral(b) => Ok((*b).into()),
-                TokenType::NumberLiteral(n) => Ok(SteelVal::NumV(*n)),
+                // todo!() figure out if ok to expand scope of eval atom.
```

Comment:
Yeah this is fine, also this function does not need to exist here, it can be removed and share the other implementation. Apologies for the shotgun of the same function over 5 places